packages/aws-cdk/test/api/logs/register-cloudwatch-logs.test.ts

```diff
@@ -0,0 +1,241 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { CloudFormation } from 'aws-sdk';
+import { registerCloudWatchLogGroups } from '../../../lib/api/logs/register-cloudwatch-logs';
+import { testStack, TestStackArtifact } from '../../util';
+import { MockSdkProvider } from '../../util/mock-sdk';
+
+let logsMockSdkProvider: LogsMockSdkProvider;
+let mockGetEndpointSuffix: () => string;
+
+beforeEach(() => {
+  logsMockSdkProvider = new LogsMockSdkProvider();
+  mockGetEndpointSuffix = jest.fn(() => 'amazonaws.com');
+  logsMockSdkProvider.stubGetEndpointSuffix(mockGetEndpointSuffix);
+  // clear the array
+  currentCfnStackResources.splice(0);
+});
+
+test('add log groups from lambda function', async () => {
+  // GIVEN
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Func: {
+          Type: 'AWS::Lambda::Function',
+          Properties: {
+            FunctionName: 'my-function',
+          },
+        },
+      },
+    },
+  });
+  pushStackResourceSummaries(stackSummaryOf('Func', 'AWS::Lambda::Function', 'my-function'));
```

Comment:
So, this is exactly the problem I mentioned above.

Even though the function's name can be established from the template, we still read it from the CloudFormation service.

If we used the `establishResourcePhysicalName()` helper in the production code, the tests would not need this line in them.

Same comment in the other tests.

Reply:
Done.